src/redisearch_rs/CMakeLists.txt

```
@@ -0,0 +1,105 @@
+cmake_minimum_required(VERSION 3.13)
+
+# Find Rust and Cargo
+find_program(CARGO_EXECUTABLE cargo REQUIRED)
+
+if(NOT CARGO_EXECUTABLE)
+    message(FATAL_ERROR "Cargo not found. Please install Rust and Cargo.")
+endif()
+
+# Determine Rust profile - allow invoker to set it, but have fallback logic if not set
+if(NOT DEFINED RUST_PROFILE)
+    set(RUST_PROFILE "release")
+    if(DEFINED BUILD_SEARCH_UNIT_TESTS AND BUILD_SEARCH_UNIT_TESTS)
+        if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR DEFINED SAN OR DEFINED COV)
+            set(RUST_PROFILE "dev")
+        else()
+            set(RUST_PROFILE "optimised_test")
+        endif()
+    else()
+        # For non-test builds, only use dev profile for Debug builds (not SAN/COV)
+        if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+            set(RUST_PROFILE "dev")
+        endif()
+    endif()
+endif()
+
+# Set up RUSTFLAGS - respect existing environment variable and default to -D warnings
+if(DEFINED ENV{RUSTFLAGS})
+    set(RUST_FLAGS "$ENV{RUSTFLAGS}")
+else()
+    set(RUST_FLAGS "-D warnings")
+endif()
+
+# Add dynamic C runtime flag if requested (crucial for Alpine Linux)
+if(DEFINED RUST_DYN_CRT AND RUST_DYN_CRT)
+    set(RUST_FLAGS "${RUST_FLAGS} -C target-feature=-crt-static")
+endif()
+
+# Exclude benchmarking crates that link C code
+set(EXCLUDE_ARGS "--exclude" "inverted_index_bencher" "--exclude" "rqe_iterators_bencher")
+
+# Map Rust profile names to their corresponding artifact directory names
+if(RUST_PROFILE STREQUAL "dev")
+    set(RUST_ARTIFACT_DIR "debug")
+else()
+    # For release, optimised_test, and other custom profiles, use the profile name as-is
+    set(RUST_ARTIFACT_DIR "${RUST_PROFILE}")
+endif()
+
+# Determine the output library path based on profile and context
+if(DEFINED BINDIR)
+    # We're being built as part of the main RediSearch build
+    set(RUST_TARGET_DIR "${BINDIR}/redisearch_rs")
+    set(RUST_LIB_PATH "${BINDIR}/redisearch_rs/${RUST_ARTIFACT_DIR}/libredisearch_rs.a")
+else()
+    # We're being built standalone or as a subdirectory
+    set(RUST_TARGET_DIR "${CMAKE_CURRENT_SOURCE_DIR}/target")
+    set(RUST_LIB_PATH "${CMAKE_CURRENT_SOURCE_DIR}/target/${RUST_ARTIFACT_DIR}/libredisearch_rs.a")
+endif()
+
+# Check if we're being used as a subdirectory to avoid target conflicts
+if(NOT TARGET redisearch_rs_build)
+    # Create the custom target for building Rust code
+    add_custom_target(redisearch_rs_build
+        COMMAND ${CMAKE_COMMAND} -E env
+            "RUSTFLAGS=${RUST_FLAGS}"
+            "CARGO_TARGET_DIR=${RUST_TARGET_DIR}"
+            ${CARGO_EXECUTABLE} build
+            --workspace
+            ${EXCLUDE_ARGS}
```

Based on the conversation above, you don't want to build the whole workspace.
You only want to build the crate that generates the static library you need, as well as its deps.
```suggestion
            ${CARGO_EXECUTABLE} build
            --package redisearch_rs
```
That should allow you to remove `EXCLUDE_ARGS` entirely.